### ISSUE TITLE
Issue parsing nested HTML elements

### ISSUE THREAD

**Issue by iprignano** [probablyup/markdown-to-jsx#255]
First of all thanks for you work on the library. We've recently adopted it and encountered an issue when trying to parse and render some markdown mixed with HTML (which is there to provide backwards compatibility with another system, hence can't be removed). 

In fact when trying to parse three nested `<div>`s:

```html
<div>
  <div>
    <div></div>
  </div>
</div>
```

`markdown-to-jsx` will output this:

```html
<div>
  <div></div>
  <div></div>
</div>
<p>&lt;/div&gt;</p>
```

It can be reproduced on the demo site, and it seems to happen when triple-nesting any HTML element. The issue looks _similar_ to https://github.com/probablyup/markdown-to-jsx/issues/168 but I'm not sure if it's exactly the same problem.

**Comment by jthistle** [comment]
I'm taking a look at this. Doubt I'll be able to fix it but we'll see :)

**Comment by iprignano** [comment]
@jthistle Thanks for your reply! 

I just wanted to add that for us this was a dealbreaker, so we had to unfortunately move to a different library. Regardless, thank you for looking into it.

**Comment by jthistle** [comment]
Ok, so I've taken a long look at this. The problem basically stems from the fact that regex is being used to parse HTML. HTML is not a regular language, and [you cannot use regex to parse it](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454). You can make it seem like it works, but it will never properly work, hence bugs like this.

I'd recommend using a real XML/HTML parser instead.

**Comment by probablyup** [comment]
True, but none of those are small enough to fit the size requirement of
this library. It will never be perfect, but it’s good enough for most use
cases.

**Comment by callmemb** [comment]
Why not skip HTML tags and let react handle those ?

**Comment by quantizor** [comment]
Should be fixed in v9
